src/taskgraph/generator.py

```diff
@@ -383,13 +383,22 @@ def _run(self):
         do_not_optimize = set(parameters.get("do_not_optimize", []))
         if not parameters.get("optimize_target_tasks", True):
             do_not_optimize = set(target_task_set.graph.nodes).union(do_not_optimize)
+
+        # this is used for testing experimental optimization strategies
+        strategies = os.environ.get(
+            "TASKGRAPH_OPTIMIZE_STRATEGIES", parameters.get("optimize_strategies")
+        )
+        if strategies:
+            strategies = find_object(strategies)
```

Comment:
nit: maybe don't overwrite variable types like this? i.e: `stategies = find_object(raw_strategies)`